project_key/models/project_project.py

```diff
@@ -138,12 +138,29 @@ def generate_project_key(self, text):
 
         data = text.split(" ")
         if len(data) == 1:
-            return data[0][:3].upper()
+            res = data[0][:3].upper()
+            return self._generate_project_unique_key(res)
 
         key = []
         for item in data:
             key.append(item[:1].upper())
-        return "".join(key)
+        res = "".join(key)
+        return self._generate_project_unique_key(res)
+
+    def _generate_project_unique_key(self, text):
+        res = text
+        unique_key = False
+        counter = 0
+        while not unique_key:
+            if counter != 0:
+                res = "%s%s" % (text, counter)
+            projects = self.search([("key", "=", res)])
+            if not projects:
+                unique_key = True
+            else:
+                counter += 1
+
```

Comment:
```suggestion
            unique_key = bool(self.search([("key", "=", res)]))
            counter += 1
```

Reply:
Thanks for the review.
I've changed it.